x-pack/legacy/plugins/security/index.js

```diff
@@ -137,28 +112,37 @@ export const security = (kibana) => new kibana.Plugin({
   },
 
   async init(server) {
-    const plugin = this;
+    const securityPlugin = this.kbnServer.newPlatform.setup.plugins.security;
+    if (!securityPlugin) {
+      throw new Error('New Platform XPack Security plugin is not available.');
+    }
 
-    const config = server.config();
     const xpackMainPlugin = server.plugins.xpack_main;
     const xpackInfo = xpackMainPlugin.info;
+    securityPlugin.registerLegacyAPI({
+      xpackInfo,
+      isSystemAPIRequest: server.plugins.kibana.systemApi.isSystemApiRequest.bind(
```

Comment:
__question:__ @restrry wondering if Core should eventually take `systemApi` over from Kibana plugin?

Reply:
seems so. as other meta-flag like `kibana-xsrf`, I will create an issue to track

Reply:
added to https://github.com/elastic/kibana/issues/39330